src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -1,62 +1,64 @@
 package seedu.address.logic.commands;
 
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
 
-import seedu.address.commons.core.Messages;
 import seedu.address.commons.core.index.Index;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.person.ReadOnlyPerson;
 import seedu.address.model.person.exceptions.PersonNotFoundException;
 
 /**
- * Deletes a person identified using it's last displayed index from the address book.
+ * Represents a command that deletes persons from the addressbook identified by some input parameter
  */
-public class DeleteCommand extends UndoableCommand {
+public abstract class DeleteCommand extends UndoableCommand {
 
     public static final String COMMAND_WORD = "delete";
     public static final String COMMAND_ALIAS = "d";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the last person listing.\n"
+            + ": Deletes the specified persons from the address book\n"
             + "Alias: " + COMMAND_ALIAS + "\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + "Options: \n"
+            + "\tdefault - Deletes the persons identified by the index numbers "
+            + "(must be positive integers) used in the last person listing.\n"
+            + "\t" + DeleteByTagCommand.COMMAND_OPTION
+            + " - Deletes the perons in the last person listing with the specified tags.\n"
+            + "Parameters: [OPTION] IDENTIFIER [MORE_IDENTIFIERS]...\n"
```

Comment:
Do we want to standardize by moving `Parameters` line just right below `Alias`? 
E.g. https://github.com/CS2103AUG2017-W10-B2/main/blob/fd8c5c7217295a787e36bb9d22bb9ec209095b9b/src/main/java/seedu/address/logic/commands/ThemeCommand.java#L18
https://github.com/CS2103AUG2017-W10-B2/main/blob/eb0b57b3b1493ed3d518ac60a2f0b80bf3aa1753/src/main/java/seedu/address/logic/commands/AddCommand.java#L27
I think it makes sense to explain the available `Options` after showing the `Parameters` too~